### PR TITLE
Update to cairo-lang v0.9.0

### LINKED ISSUE #130: Update to Cairo 0.9.0

### PR DESCRIPTION
**Pull request by andrew-fleming**
Supercedes #131, fixes #130 and #96.

This PR proposes to:
- add `max_fee` to the tx flow
- include `declare` command in CLI and NRE
  - tracks declarations in a `declarations.txt`
- update cairo-lang to v.0.9.0 in tox dependencies
- update signer to use `contract_class` instead of `contract_def`
- fix host in `clean`

### CODE REVIEW

**Review by martriay — CHANGES_REQUESTED** [review]
Looking good! Left a few comments

File: src/nile/nre.py [hunk 1]
```diff
@@ -38,9 +43,13 @@ def invoke(self, contract, method, params=None):
             params = []
         return call_or_invoke(contract, "invoke", method, params, self.network)
 
-    def get_deployment(self, contract):
+    def get_deployment(self, identifier):
         """Get a deployment by its identifier (address or alias)."""
-        return next(deployments.load(contract, self.network))
+        return next(deployments.load(identifier, self.network))
+
+    def get_declaration(self, identifier):
```
[the remark this answers is not in the document]
Comment (andrew-fleming) [reply src/nile/nre.py:50]: If I understand your question correctly, it's usable in the NRE. Users can declare a class in the script and just use the returned hash, but I'm thinking this adds flexibility if the class is already declared in their `declarations.txt`

File: src/nile/core/call_or_invoke.py [hunk 2]
```diff
@@ -37,4 +39,9 @@ def call_or_invoke(contract, type, method, params, network, signature=None):
         command.append("--signature")
         command.extend(signature)
 
-    return subprocess.check_output(command).strip().decode("utf-8")
+    if max_fee is not None:
+        command.append("--max_fee")
+        command.append(max_fee)
+
+    output = subprocess.check_output(command).strip().decode("utf-8")
```
[the remark this answers is not in the document]
Comment (andrew-fleming) [reply src/nile/core/call_or_invoke.py:46]: Yes, the `max_fee` will always be included (set to `0` if not included by users). Even if we didn't set it automatically, there's latency between sending and rejecting a tx; thus, an exception won't be raised here

File: README.md [hunk 3]
```diff
@@ -146,41 +146,54 @@ A few things to notice here:
 
 ### `send`
 
-Execute a transaction through the `Account` associated with the private key provided. The syntax is:
+Execute a transaction through the `Account` associated with the private key provided.  The syntax is:
```
Comment (martriay) [inline README.md:149]:
```suggestion
Execute a transaction through the `Account` associated with the private key provided. The syntax is:
```

File: src/nile/nre.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrew-fleming) [reply src/nile/nre.py:50]: Good call

File: src/nile/deployments.py [hunk 4]
```diff
@@ -29,10 +29,8 @@ def register_class_hash(hash, network, alias):
     """Register a new deployment."""
     file = f"{network}.{DECLARATIONS_FILENAME}"
 
-    if class_exists(hash, network, alias):
-        raise Exception(
-            f"Duplicate hash {hash[:6]}...{hash[-6:]} or alias {alias} in {file}"
-        )
+    if hash_exists(hash, network):
```
Comment (martriay) [inline src/nile/deployments.py:32]: i liked the term _class_, although I see how the class could be represented by an alias. in any case "hash" is a bit generic in any crypto context 😅 

i vote for either `class_exists` or `class_hash_exists`, or similar

File: src/nile/deployments.py [hunk 5]
```diff
@@ -29,10 +29,8 @@ def register_class_hash(hash, network, alias):
...
+        raise Exception(f"Hash {hash[:6]}...{hash[-6:]} already exists in {file}")
```
Comment (martriay) [inline src/nile/deployments.py:33]: i like this message better

**Review by martriay — APPROVED** [review]
Very good! Left few comments, I also think we should document `get_declaration`.

Otherwise, good to go!

**Pull request merged**